pingouin/contingency.py

```diff
@@ -0,0 +1,177 @@
+    high :math:`\chi^2` and low p-value. In the second example, similar
+    categorical variables should present low :math:`\chi^2` and high p-value.
+
+    .. warning :: As a general guideline for the consistency of this test, the
+        observed and the expected contingency tables should not have cells
+        with frequencies lower than 5.
+
+    References
+    ----------
+    .. [1] Cressie, N., & Timothy R. C. Read. (1984). Multinomial
+       Goodness-of-Fit Tests. Journal of the Royal Statistical Society. Series
+       B (Methodological), 46(3), 440-464.
+
+    .. [2] Yates, F. (1934). Contingency Tables Involving Small Numbers and the
+       :math:`\chi^2` Test. Supplement to the Journal of the Royal Statistical
+       Society, 1, 217-235.
+
+    Examples
+    --------
+    Let's see if gender is a good categorical predictor for the presence of
+    heart disease.
+
+    >>> import pingouin as pg
+    >>> data = pg.read_dataset('heart')
+    >>> data['sex'].value_counts(ascending=True)
+    0     96
+    1    207
+    Name: sex, dtype: int64
+
+    If gender is not a good predictor for heart disease, we should expect the
+    same 96:207 ratio across the target classes.
+
+    >>> expected, observed, dof, stats = pg.chi2(data, 'sex', 'target')
+    >>> expected
+    target          0           1
+    sex
+    0       43.722772   52.277228
+    1       94.277228  112.722772
+
+    Let's see what the data tells us.
+
+    >>> dof
+    1
+
+    The observed contingency table was adjusted. We should see fractional
+    cells.
+
+    >>> observed
+    target      0     1
+    sex
+    0        24.5  71.5
+    1       113.5  93.5
+
+    The proportion is lower on the class 0 and higher on the class 1. The
+    tests should be sensitive to this difference.
+
+    >>> stats
+         lambda       chi2             p
+    0  1.000000  22.717227  1.876778e-06
+    1  0.666667  22.931427  1.678845e-06
+    2  0.000000  23.557374  1.212439e-06
+    3 -0.500000  24.219622  8.595211e-07
+    4 -1.000000  25.071078  5.525544e-07
+    5 -2.000000  27.457956  1.605471e-07
+
+    Very low p-values indeed. The gender qualifies as a good predictor for the
+    presence of heart disease on this dataset.
+    """
+    # Python code inspired by SciPy's chi2_contingency
+    assert isinstance(data, pd.DataFrame), 'data must be a pandas DataFrame.'
+    assert isinstance(x, str), 'x must be a string.'
+    assert isinstance(y, str), 'y must be a string.'
+    assert all(col in data.columns for col in (x, y)),\
+        'columns are not in dataframe.'
+    assert isinstance(correction, bool), 'correction must be a boolean.'
+
+    observed = pd.crosstab(data[x], data[y])
+
+    if observed.size == 0:
+        raise ValueError('No data; observed has size 0.')
+
+    expected = pd.DataFrame(expected_freq(observed), index=observed.index,
+                            columns=observed.columns)
+
+    expected_zero_mask = expected == 0
```

Comment:
I just realized that `expected` can't have cells with values 0 because it was built with `pd.crosstab`